test/FieldTests/RandomiseMines.java

```diff
@@ -8,7 +8,7 @@
 
     @Test
     public void ifAFieldIsCreatedWithRandomMinePlacementTheFieldWillNotBeEmpty() {
-        Field field = new Field(5,5, new MockMineRandom(1, 0, 0));
+        Field field = new Field(5,5, new RandomNumberGenerator());
```

Comment:
This test isn't really needed. This could also fail, and remember, we never want 'flaky' tests that can sometimes fail. Your existing tests that use your `PredictableNumberGenerator` are all of the test coverage you need on the randomisation method. This might feel weird because you've actually never tested it with properly random numbers, but that's ok.

If you want to test your random number generator, you'll probably need to provide a way for it to accept a seed, and then write some tests that assert that given a particular seed, when you call `nextInt` with some arguments it gives you back a particular value.

Reply:
I was thinking there would have to be some sort of test using the the real RandomNumberGenerator or how do you really know it was working. Then again that is why you use the predictable one.